File: src/main/java/com/jaasielsilva/helpdesk/model/Usuario.java
package com.jaasielsilva.helpdesk.model;

import java.time.LocalDateTime;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import com.jaasielsilva.helpdesk.enums.PerfilUsuario;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import jakarta.persistence.Version;
import lombok.Data;

@Data
@Entity
@Table(name = "usuarios", uniqueConstraints = {
        @UniqueConstraint(name = "uk_usuario_empresa", columnNames = { "empresa_id", "usuario" })
})
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Version
    private Long version;

    private String nome;

    @Column(nullable = false)
    private String usuario;

    @Column(nullable = false)
    private String senha;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PerfilUsuario perfil = PerfilUsuario.USER;

    @ManyToOne
    @JoinColumn(name = "empresa_id")
    private Empresa empresa;

    @Column(nullable = false)
    private boolean ativo = true;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime dataCriacao;

    @UpdateTimestamp
    private LocalDateTime dataAtualizacao;

    private LocalDateTime deletedAt;

    public boolean isSuperAdmin() {
        return perfil == PerfilUsuario.SUPER_ADMIN;
    }

    public boolean belongsToEmpresa(Long empresaId) {
        return empresa != null && empresa.getId().equals(empresaId);
    }
}

File: frontend/src/app/core/models/usuario.ts
import { UserRole } from './user-role';

export interface Usuario {
  id: number;
  usuario: string;
  nome: string;
  perfil: UserRole;
  empresaId: number | null;
  empresaNome: string | null;
  empresaSlug: string | null;
  ativo: boolean;
  dataCriacao: string;
}

export interface CreateUsuarioRequest {
  usuario: string;
  senha: string;
  nome?: string;
  perfil: UserRole;
  empresaId?: number | null;
}

export interface UpdateUsuarioRequest {
  nome?: string;
  perfil?: UserRole;
  ativo?: boolean;
  senha?: string;
}

export interface PageResponse<T> {
  content: T[];
  page: {
    totalElements: number;
    totalPages: number;
    number: number;
    size: number;
  };
}
